### PR TITLE
8322751: ZGC: Fix comments about marking roots

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

This trivial patch fixes comments about marking roots.

Thread-stack processing has been concurrent after [JDK-8253180](https://bugs.openjdk.org/browse/JDK-8253180).
And the code about marking root in `ZMark::start` has been removed after [JDK-8254562](https://bugs.openjdk.org/browse/JDK-8254562).
But some comments in ZGC and generational ZGC have not fixed, which may mislead the newbies of the ZGC.

Thanks for review.

Best Regards,
-- Guoxiong

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8322751](https://bugs.openjdk.org/browse/JDK-8322751): ZGC: Fix comments about marking roots (**Enhancement** - P5)

### Reviewers
 * [Erik Österlund](https://openjdk.org/census#eosterlund) (@fisk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17195/head:pull/17195` \
`$ git checkout pull/17195`

Update a local copy of the PR: \
`$ git checkout pull/17195` \
`$ git pull https://git.openjdk.org/jdk.git pull/17195/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17195`

View PR using the GUI difftool: \
`$ git pr show -t 17195`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17195.diff">https://git.openjdk.org/jdk/pull/17195.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17195#issuecomment-1870092870)

### CODE REVIEW

**Review by fisk — APPROVED** [review]

**Comment by lgxbslgx** [comment]
@fisk Thanks for your review. Do you think this patch is trivial? If not, I will wait for another review.

**Comment by fisk** [comment]
It looks trivial; ship it!

**Pull request closed without merging**
